src/lib/parser/statement/splitter.rs

```diff
@@ -109,18 +120,33 @@ impl<'a> StatementSplitter {
             math_paren_level: 0,
         }
     }
+
+    fn get_statement(&mut self, flags: Flags) -> StatementVariant<'a> {
+        self.flags.insert(flags);
+        match self.flags {
```

Comment:
This also won't work because flags contains many flags. You'll need to switch to `if flags.contains(Flags::AND) / else if flags.contains(Flags::OR) / else`

Reply:
Will do